PactNet.Tests/PactUriOptionsTests.cs

```diff
@@ -5,31 +5,93 @@ namespace PactNet.Tests
 {
     public class PactUriOptionsTests
     {
+
+        // This is an experimental test set.
+        // Use of Data Driving improves readability and 
+        // shows all the permutations run by one test
+        [Theory]

```

Comment:
I am pretty familiar with this kind of test cases concept although I am using Nunit. However I feel like your approach is not correct due to:
 
 1. You test combination of user name and password, while that is not case in code. It will be enough to check not valid name and not empty password separately. IMHO you just multiple number of cases, which are not adding anything new.
 2. You duplicate a exception message e.g.  `username contains a ':' character, which is not allowed`
 3. You tests are a bit less descriptive.
 
 ```csharp
  private const string emptyUserNameErrorMessage = "username is null or empty.";

        [Theory]
        [InlineData("bad:name", "username contains a ':' character, which is not allowed.")]
        [InlineData("", emptyUserNameErrorMessage)]
        [InlineData(null, emptyUserNameErrorMessage)]
        public void Ctor_WhenUserNameIsNotAcceptable_ShouldThrowArgumentException(string notAcceptableUserName, string expectedMessage)
        {
            var exception = Assert.Throws<ArgumentException>(() => new PactUriOptions(notAcceptableUserName, "dummyPassword"));

            Assert.Equal(expectedMessage, exception.Message);
        }

        [Theory]
        [InlineData("")]
        [InlineData(null)]
        public void Ctor_WhenPasswordIsNotProvided_ShouldThrowArgumentException(string password)
        {
            var exception = Assert.Throws<ArgumentException>(() => new PactUriOptions("dummyUserName", password));

            Assert.Equal("password is null or empty.", exception.Message);
        }

```

Reply:
The trade-off with data driving is always the lower descriptiveness. But I concur, I think I am treating this like black box tests

Reply:
I have split these as you have suggested, and commented where there is RFC 2716 compliance